### PR TITLE
Fix rubocop warning for KCC servicemapping provider

### PR DESCRIPTION
**Pull request by rileykarson**
There was a copyright error in the original PR (https://github.com/GoogleCloudPlatform/magic-modules/pull/6970) that was gonna get fixed by just merging; ends up there's a later round of rake tests contingent on success.

fyi @maqiuyujoyce. Most of these are automated fixes, I'll flag the couple that aren't. You may want to reverify the output.

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
```

### CODE REVIEW

File: mmv1/provider/terraform_kcc.rb [hunk 1]
```diff
@@ -87,76 +87,100 @@ def copy_common_files(output_folder, generate_code, generate_docs) end
...
     def get_resource_id_value_template(id_template, is_server_generated_name, object)
-      return nil if !is_server_generated_name
+      return nil unless is_server_generated_name
 
       if id_template.split('/').length == 1 && object.base_url != id_template
         raw_value_template = object.base_url
       end
-      return nil if raw_value_template == nil
 
-      value_template = raw_value_template + '/{{value}}'
```
Comment (rileykarson) [inline mmv1/provider/terraform_kcc.rb:118]: This change was manual- setting `value_template` didn't do anything so Rubocop wanted it removed. 116 and 122 are the same line (just adjusted to call `.nil?` automatically)

File: mmv1/provider/terraform_kcc.rb [hunk 2]
```diff
@@ -87,76 +87,100 @@ def copy_common_files(output_folder, generate_code, generate_docs) end
...
+        import_id_formats_from_resource(object)[-1].split('/')[-1].scan(/{{[[:word:]]+}}/)
       # If it is a value field, the length of last_import_part will be 1;
       # otherwise it'll be 0.
       # Remove '{{' and '}}' and only return the field name.
       last_import_part.first.gsub('{{', '').gsub('}}', '') if last_import_part.length == 1
     end
 
-    def is_server_generated_name(name, object)
-      has_computed_name_configured = object.custom_code.post_create == 'templates/terraform/post_create/set_computed_name.erb' if object.custom_code.post_create
+    def server_generated_name?(name, object)
```
Comment (rileykarson) [inline mmv1/provider/terraform_kcc.rb:98]: This name change was manual. This is called from the template, so that was changed as well.

**Review by slevenick — APPROVED** [review]
LGTM.

Do we have a way to see if there are diffs in the generated servicemapping?

**Comment by rileykarson** [comment]
Not ATM- we'd discussed hooking it up to the Magician but opted not to yet. I believe this is still in the prototype stage on the KCC side.

File: mmv1/provider/terraform_kcc.rb (hunk 1 above)
Comment (maqiuyujoyce) [reply mmv1/provider/terraform_kcc.rb:118]: Actually this line is necessary. I'm adding it back in https://github.com/GoogleCloudPlatform/magic-modules/pull/7123

File: mmv1/provider/terraform_kcc.rb (hunk 2 above)
Comment (maqiuyujoyce) [reply mmv1/provider/terraform_kcc.rb:98]: Looks good, thx!

**Pull request merged**